tests/vanilla/utils/atomWithReset.test.ts

```diff
@@ -0,0 +1,48 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { RESET, atomWithReset } from 'jotai/vanilla/utils'
+
+vi.mock('jotai/vanilla', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('jotai/vanilla')>()
+  return {
+    ...actual,
+    atom: vi.fn(actual.atom),
+  }
+})
```

Comment:
we haven't don this before, but it looks nice.

Reply:
I have remove it, it is useless in this case.